src/megaapi_impl.cpp

```diff
@@ -19918,6 +19810,9 @@ void MegaApiImpl::sendPendingRequests()
             {
                 client->locallogout(false);
                 client->restag = nextTag;
+#ifdef ENABLE_SYNC
+                mCachedMegaSyncPrivate.reset();
```

Comment:
It is probably tidier to reset this member in the `logout_result()`, as the rest of members.
I'll move it there